src/clj/db.clj

```diff
@@ -494,7 +502,47 @@
                   :where
                   [?e :user/git-id ?git-id]
                   [?e :user/id ?id]]
-                @(.conn this) git-id))))
+                @(.conn this) git-id)))
+
+
+  (get-all-corrections-from-user
+    [this user-id]
+    (mapv
+      #(zipmap [:correction/feedback :answer/points :question/points :question/question-statement :correction/timestamp :answer/answer] %)
+      (d/q '[:find ?feedback ?points-reached ?reachable-points ?question-statement ?timestamp ?answers
+             :in $ ?user-id
+             :where
+             [?correction :correction/answer ?answer]
+             [?correction :correction/feedback ?feedback ?tx]
+             [?answer :answer/user ?user]
+             [?answer :answer/points ?points-reached]
+             [?answer :answer/answer ?answers]
+             [?answer :answer/question ?question]
+             [?question :question/question-statement ?question-statement]
+             [?question :question/points ?reachable-points]
+             [?tx :db/txInstant ?timestamp]
+             [?user :user/id ?user-id]]
+           @(.conn this) user-id)))
+
+
+  (get-all-corrections-from-corrector
+    [this corrector-id]
+    (mapv
+      #(zipmap [:correction/feedback :answer/points :question/points :question/question-statement :correction/timestamp :answer/answer] %)
+      (d/q '[:find ?feedback ?points-reached ?reachable-points ?question-statement ?timestamp ?answers
+             :in $ ?user-id
+             :where
+             [?correction :correction/answer ?answer]
+             [?correction :correction/feedback ?feedback ?tx]
+             [?correction :correction/corrector ?user]
+             [?answer :answer/points ?points-reached]
+             [?answer :answer/answer ?answers]
+             [?answer :answer/question ?question]
+             [?question :question/question-statement ?question-statement]
+             [?question :question/points ?reachable-points]
+             [?tx :db/txInstant ?timestamp]
+             [?user :user/id ?user-id]]
```

Comment:
Same here. Not sure if there is a better order for this.

Reply:
I completely forgot to think about the order on these queries, thank you for the hint :D
I updated them, what do you think about them now?

Reply:
I think the most import one is `[?user :user/id ?user-id] at the top. 👍